delfin/drivers/dell_emc/unity/unity.py

```diff
@@ -650,3 +703,272 @@ def clear_alert(self, context, alert):
+                    start_time,
+                    end_time)
+                UnityStorDriver.count_total_perf(volume_metrics)
+                UnityStorDriver.package_metrics(storage_id,
+                                                constants.ResourceType.VOLUME,
+                                                metrics, volume_metrics)
+            if resource_metrics.get(constants.ResourceType.DISK):
+                disk_metrics = self.get_history_metrics(
+                    constants.ResourceType.DISK,
+                    resource_metrics.get(constants.ResourceType.DISK),
+                    start_time,
+                    end_time)
+                UnityStorDriver.count_total_perf(disk_metrics)
+                UnityStorDriver.package_metrics(storage_id,
+                                                constants.ResourceType.DISK,
+                                                metrics, disk_metrics)
+            if resource_metrics.get(constants.ResourceType.PORT):
+                port_metrics = self.get_history_metrics(
+                    constants.ResourceType.PORT,
+                    resource_metrics.get(constants.ResourceType.PORT),
+                    start_time,
+                    end_time)
+                UnityStorDriver.count_total_perf(port_metrics)
+                UnityStorDriver.package_metrics(storage_id,
+                                                constants.ResourceType.PORT,
+                                                metrics, port_metrics)
+            if resource_metrics.get(constants.ResourceType.FILESYSTEM):
+                file_metrics = self.get_history_metrics(
+                    constants.ResourceType.FILESYSTEM,
+                    resource_metrics.get(constants.ResourceType.FILESYSTEM),
+                    start_time,
+                    end_time)
+                UnityStorDriver.count_total_perf(file_metrics)
+                UnityStorDriver.package_metrics(
+                    storage_id, constants.ResourceType.FILESYSTEM,
+                    metrics, file_metrics)
+        except Exception as err:
+            err_msg = "Failed to collect metrics from Unity: %s" % \
+                      (six.text_type(err))
+            LOG.error(err_msg)
+            raise exception.InvalidResults(err_msg)
+        return metrics
+
+    @staticmethod
+    def get_capabilities(context, filters=None):
+        """Get capability of supported driver"""
+        return {
+            'is_historic': True,
+            'resource_metrics': {
+                constants.ResourceType.VOLUME: consts.VOLUME_CAP,
+                constants.ResourceType.PORT: consts.PORT_CAP,
+                constants.ResourceType.DISK: consts.DISK_CAP,
+                constants.ResourceType.FILESYSTEM: consts.FILESYSTEM_CAP
+            }
+        }
+
+    def get_latest_perf_timestamp(self, context):
+        latest_time = 0
+        page = 1
+        results = self.rest_handler.get_history_metrics(
```

Comment:
Can I obtain the maximum performance collection time of a device when the value of page is 1?

Reply:
yes，the performance data is Descending order，the earliest data in the first page